### PR TITLE
CB-4344 fix: table column width calculation

### PR DESCRIPTION
**Pull request by Wroud**

### CODE REVIEW

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          width =>
+            width + COLUMN_PADDING + COLUMN_HEADER_ICON_WIDTH + COLUMN_HEADER_TEXT_PADDING + COLUMN_HEADER_ORDER_PADDING + COLUMN_HEADER_ORDER_WIDTH,
+        );
+
+        const cellsWidth = TextTools.getWidth({
+          font: FONT,
+          text: rowStrings,
+        }).map(width => width + COLUMN_PADDING);
+
+        const width = Math.min(300, Math.max(columnsWidth[0], cellsWidth[0] ?? 0));
```
Comment (sergeyteleshev) [inline webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts]: no magic numbers

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// TODO: clear removed columns from cache
+export function useTableDataMeasurements(model: IDatabaseDataModel<any, IDatabaseResultSet>, resultIndex: number): ITableDataMeasurements {
+  const format = model.source.getAction(resultIndex, ResultSetFormatAction);
+  const view = model.source.getAction(resultIndex, ResultSetViewAction);
+
+  return useObservableRef(
+    () => ({
+      cache: observable.map(),
+      getColumnWidth(key: IResultSetColumnKey): number {
+        return this.cache.get(ResultSetDataKeysUtils.serialize(key)) ?? 300;
```
Comment (sergeyteleshev) [inline webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts]: no magic numbers

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader.tsx [hunk 3]
```diff
@@ -65,6 +65,12 @@ export const TableColumnHeader = observer<RenderHeaderCellProps<any>>(function T
     dataGridContext.focus();
   }
 
+  useMemo(() => {
+    if (calculatedColumn.columnDataIndex) {
+      calculatedColumn.onRenderHeader?.(calculatedColumn.columnDataIndex);
+    }
+  }, [calculatedColumn, calculatedColumn.columnDataIndex]);
```
Comment (sergeyteleshev) [inline webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader.tsx]: no need for `calculatedColumn.columnDataIndex` dep

**Review by sergeyteleshev — APPROVED** [review]

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts (hunk 1 above)
Comment (Wroud) [reply webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts]: fixed

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts (hunk 2 above)
Comment (Wroud) [reply webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableDataMeasurements.ts]: fixed

File: webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader.tsx (hunk 3 above)
Comment (Wroud) [reply webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader.tsx]: thanks removed

**Review by devnaumov — APPROVED** [review]

**Pull request merged**